### PR TITLE
[CoreML] Add support for xcode 14 beta 1, 2 & 3

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/coreml.cs [hunk 1]
```diff
@@ -418,12 +419,28 @@ interface MLModel {
...
+		[Static]
+		[Export ("loadModelAsset:configuration:completionHandler:")]
+		void Load (MLModelAsset asset, MLModelConfiguration configuration, Action<MLModel, NSError> handler);
+
 		// Category MLModel (MLModelCompilation)
 
+		[Deprecated (PlatformName.MacOSX, 13, 0, message: "Use the asynchronous method instead.")]
+		[Deprecated (PlatformName.iOS, 16, 0, message: "Use the asynchronous interface instead.")]
+		[Deprecated (PlatformName.TvOS, 16, 0, message: "Use the asynchronous interface instead.")]
+		[Deprecated (PlatformName.MacCatalyst, 16, 0, message: "Use the asynchronous interface instead.")]
```
Comment (dalexsoto) [inline src/coreml.cs:433]:
```suggestion
		[Deprecated (PlatformName.MacOSX, 13, 0, message: "Use 'CompileModel (NSUrl, Action<NSUrl, NSError>)' overload or 'CompileModelAsync' instead.")]
		[Deprecated (PlatformName.iOS, 16, 0, message: "Use 'CompileModel (NSUrl, Action<NSUrl, NSError>)' overload or 'CompileModelAsync' instead.")]
		[Deprecated (PlatformName.TvOS, 16, 0, message: "Use 'CompileModel (NSUrl, Action<NSUrl, NSError>)' overload or 'CompileModelAsync' instead.")]
		[Deprecated (PlatformName.MacCatalyst, 16, 0, message: "Use 'CompileModel (NSUrl, Action<NSUrl, NSError>)' overload or 'CompileModelAsync' instead.")]
```

File: src/coreml.cs [hunk 2]
```diff
@@ -507,6 +524,11 @@ interface MLModelMetadata {
 	[BaseType (typeof (NSObject))]
 	interface MLMultiArray : NSSecureCoding {
 
+		[Deprecated (PlatformName.MacOSX, 13, 0, message: "Use 'GetBytesWithHandler' or 'GetMutableBytes' async methods instead.")]
+		[Deprecated (PlatformName.iOS, 16, 0, message: "Use 'GetBytesWithHandler' or 'GetMutableBytes' async methods instead.")]
+		[Deprecated (PlatformName.TvOS, 16, 0, message: "Use 'GetBytesWithHandler' or 'GetMutableBytes' async methods instead.")]
+		[Deprecated (PlatformName.WatchOS, 9, 0, message: "Use 'GetBytesWithHandler' or 'GetMutableBytes' async methods instead.")]
+		[Deprecated (PlatformName.MacCatalyst, 16, 0, message: "Use the asynchronous interface instead.")]
```
Comment (dalexsoto) [inline src/coreml.cs:531]: Why is this message different to the ones above?

File: src/coreml.cs [hunk 3]
```diff
@@ -507,6 +524,11 @@ interface MLModelMetadata {
 	[BaseType (typeof (NSObject))]
 	interface MLMultiArray : NSSecureCoding {
 
+		[Deprecated (PlatformName.MacOSX, 13, 0, message: "Use 'GetBytesWithHandler' or 'GetMutableBytes' async methods instead.")]
+		[Deprecated (PlatformName.iOS, 16, 0, message: "Use 'GetBytesWithHandler' or 'GetMutableBytes' async methods instead.")]
+		[Deprecated (PlatformName.TvOS, 16, 0, message: "Use 'GetBytesWithHandler' or 'GetMutableBytes' async methods instead.")]
+		[Deprecated (PlatformName.WatchOS, 9, 0, message: "Use 'GetBytesWithHandler' or 'GetMutableBytes' async methods instead.")]
```
Comment (dalexsoto) [inline src/coreml.cs:530]:
```suggestion
		[Deprecated (PlatformName.MacOSX, 13, 0, message: "Use 'GetBytes (Action<IntPtr, nint>)' or 'GetMutableBytes' async methods instead.")]
		[Deprecated (PlatformName.iOS, 16, 0, message: "Use 'GetBytes (Action<IntPtr, nint>)' or 'GetMutableBytes' async methods instead.")]
		[Deprecated (PlatformName.TvOS, 16, 0, message: "Use 'GetBytes (Action<IntPtr, nint>)' or 'GetMutableBytes' async methods instead.")]
		[Deprecated (PlatformName.WatchOS, 9, 0, message: "Use 'GetBytes (Action<IntPtr, nint>)' or 'GetMutableBytes' async methods instead.")]
```

File: src/coreml.cs [hunk 4]
```diff
@@ -1097,4 +1133,16 @@ interface MLModelCollectionEntry {
 		[Export ("isEqualToModelCollectionEntry:")]
 		bool IsEqual (MLModelCollectionEntry entry);
 	}
+
+	[Watch (9,0), TV (16,0), Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
```
Comment (dalexsoto) [inline src/coreml.cs:1137]: `Watch (9,0)` ? The type seems to be useless, the only method seems to be "NoWatch"

**Review by dalexsoto — CHANGES_REQUESTED** [review]

File: src/coreml.cs (hunk 2 above)
Comment (mandel-macaque) [reply src/coreml.cs:531]: Cause I yy and missed it

Comment (mandel-macaque) [inline src/coreml.cs:531]:
```suggestion
		[Deprecated (PlatformName.MacCatalyst, 16, 0, message: "Use 'GetBytes (Action<IntPtr, nint>)' or 'GetMutableBytes' async methods instead.")]
```

File: src/coreml.cs (hunk 4 above)
Comment (mandel-macaque) [reply src/coreml.cs:1137]: Seems to be needed for loadModelAsset:configuration:completionHandler: which is present on watchOS. The model asset can be loaded form a file. Maybe there is an API we are missing about that.

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**